src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -156,7 +150,15 @@ private void EnableTimer(float pollingIntervalInSeconds)
                         restoreFlow = true;
                     }
 
-                    _pollingTimer = new Timer(s => ((CounterGroup)s!).OnTimer(null), this, _pollingIntervalInMilliseconds, _pollingIntervalInMilliseconds);
+                    if (_pollingThread == null)
+                    {
+                        _pollingThread = new Thread(() => PollForValues());
```

Comment:
You do not  need to wrap the PollForValues in a lambda. It can be passed as delegate directly.